examples/http-header-filter/cafe-route.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /coffee
+    filters:
+    - type: RequestHeaderModifier
+      requestHeaderModifier:
+        set:
+        - name: My-Overwrite-Header
+          value: "this-is-the-only-value"
```

Comment:
do we really need to use ".." for header values? let's not use them if they are not necessary, since we don't use them for other fields